packages/webapp/src/components/Task/AddSoilAmendmentProducts/index.tsx

```diff
@@ -0,0 +1,140 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { useMemo, useState } from 'react';
+import { useFieldArray, useFormContext } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import useExpandable from '../../Expandable/useExpandableItem';
+import ProductCard, { type ProductCardProps } from './ProductCard';
+import TextButton from '../../Form/Button/TextButton';
+import { type Product } from './types';
+import { defaultValues } from './ProductCard/ProductDetails';
+import { ReactComponent as PlusCircleIcon } from '../../../assets/images/plus-circle.svg';
+import { TASK_TYPES } from '../../../containers/Task/constants';
+import styles from './styles.module.scss';
+
+export type AddSoilAmendmentProductsProps = ProductCardProps & { products: Product[] };
+
+type ProductId = 'number' | 'string';
+
+interface ProductFields {
+  product_id: ProductId;
+}
+
+const FIELD_NAME = 'soil_amendment_task_products';
+
+const AddSoilAmendmentProducts = ({
+  products,
+  isReadOnly,
+  ...props
+}: AddSoilAmendmentProductsProps) => {
+  const [invalidProducts, setInvalidProducts] = useState<string[]>([]);
+
+  const { t } = useTranslation();
+  const { control, setValue, watch } = useFormContext();
+  const { fields, append, remove } = useFieldArray({
+    name: FIELD_NAME,
+    control,
+  });
+
+  const productsOfType = useMemo(
```

Comment:
I think this was already here from existing add product?  I always think about: https://react.dev/reference/react/useMemo#how-to-tell-if-a-calculation-is-expensive and wonder if we can remove the memo.

Reply:
I guess all of the above haha. I am just trusting the React docs that unless looping over 1000's of items its not worth it. 🤷

Reply:
The doc also says "There is no significant harm to doing that" so I was wondering.
I don't like that the filtering keeps happening when products don't change, but it doesn't affect the performance so I'm okay without useMemo.
I wish we could filter the product somewhere else!

Reply:
> I wish we could filter the product somewhere else!

How about in a new selector in `productSlice`? 😄 

I think that would end up equivalent to `useMemo()` but with Redux Toolkit handling the memoization. I might be on a selector kick, but I think that would be readability without recalculating.

Reply:
That's what I wanted to do! I would need to update three files, so I hesitated to go for it.
But now that you mentioned, I will create a new selector 😄